nbgitpuller/pull.py

```diff
@@ -43,23 +43,25 @@ def flush():
             raise subprocess.CalledProcessError(ret, cmd)
 
 class GitPuller(Configurable):
-    depth = Integer(None, allow_none=True, config=True,
-                    help="""
-                    Depth (ie, commit count) to which to perform a
-                    shallow git clone.
+    depth = Integer(
+        int(os.environ.get('NBGITPULLER_DEPTH', 1)),
+        config=True,
+        help="""
+        Depth (ie, commit count) to which to perform a
+        shallow git clone.
 
-                    If not set, disables shallow clones.
+        If not set, clones to depth 1.
 
-                    Defaults to reading from the NBGITPULLER_DEPTH
-                    environment variable.
-                    """)
+        Defaults to reading from the NBGITPULLER_DEPTH
+        environment variable."""
+    )
 
     @default('depth')
     def _depth_default(self):
         depth = os.environ.get('NBGITPULLER_DEPTH')
         if depth:
             return int(depth)
-        return None
+        return 1
```

Comment:
I think this function does the same as, but made me confused that it may do something more fancy.

```
return os.environ.get('NBGITPULLER_DEPTH', 1)
```

Reply:
Good catch! I removed the function - it isn't needed at all anymore

Reply:
I'm wondering if we should keep the dynamic default after all?
I believe the `test_environment_shallow_clone` fails because we're only initializing the depth with the env var value, but if that gets changed, nothing happens with depth, right?